Sources/xcodes/main.swift

```diff
@@ -217,69 +229,103 @@ struct Xcodes: ParsableCommand {
+
             var downloader = XcodeInstaller.Downloader.urlSession
             if let aria2Path = aria2.flatMap(Path.init) ?? Current.shell.findExecutable("aria2c"),
                aria2Path.exists,
                noAria2 == false {
                 downloader = .aria2(aria2Path)
             }
-            
+
             let destination = getDirectory(possibleDirectory: directory)
-            
-            installer.install(installation, dataSource: globalDataSource.dataSource, downloader: downloader, destination: destination, experimentalUnxip: experimentalUnxip, emptyTrash: emptyTrash, noSuperuser: noSuperuser)
-                .done { Install.exit() }
-                .catch { error in
-                    Install.processDownloadOrInstall(error: error)
+
+            if select, case .version(let version) = installation {
+                firstly {
+                    selectXcode(shouldPrint: print, pathOrVersion: version, directory: destination, fallbackToInteractive: false)
+                }
+                .catch { _ in
+                    install(installation, using: downloader, to: destination)
                 }
-            
+            } else {
+                install(installation, using: downloader, to: destination)
+            }
+
             RunLoop.current.run()
         }
+
+        private func install(_ installation: XcodeInstaller.InstallationType,
+                             using downloader: XcodeInstaller.Downloader,
+                             to destination: Path) {
+            firstly { () -> Promise<InstalledXcode> in
+                // update the list before installing only for version type because the other types already update internally
+                if update, case .version = installation {
+                    Current.logging.log("Updating...")
+                    return xcodeList.update(dataSource: globalDataSource.dataSource)
+                        .then { _ -> Promise<InstalledXcode> in
+                            installer.install(installation, dataSource: globalDataSource.dataSource, downloader: downloader, destination: destination, experimentalUnxip: experimentalUnxip, emptyTrash: emptyTrash, noSuperuser: noSuperuser, exitAfterInstall: !select)
+                        }
+                } else {
+                    return installer.install(installation, dataSource: globalDataSource.dataSource, downloader: downloader, destination: destination, experimentalUnxip: experimentalUnxip, emptyTrash: emptyTrash, noSuperuser: noSuperuser, exitAfterInstall: !select)
+                }
+            }
+            .then { xcode -> Promise<Void> in
+                Current.logging.log("\nXcode \(xcode.version.descriptionWithoutBuildMetadata) has been installed to \(xcode.path.string)".green)
+
+                // Install was successful, now select it
+                return selectXcode(shouldPrint: print, pathOrVersion: xcode.path.string, directory: destination, fallbackToInteractive: false)
```

Comment:
The `selectXcode` here isn't looking at the `select` option passed in.

Reply:
Actually, this line will not get executed if select is false because install would exit early from the step before. That's why I don't have a check here.

Reply:
Unless I missed something in how Install.exit works.

Reply:
Yep I tested it out

```
➜  xcodes git:(add-select-option-to-install) swift run xcodes install 14.0.1 --experimental-unxip
Building for debugging...
Build complete! (0.08s)

Downloading with aria2
(1/6) Downloading Xcode 14.0.1+14A400: 99%
(2/6) Unarchiving Xcode (This can take a while)
Using experimental unxip. If you encounter any issues, remove the flag and try again
(3/6) Moving Xcode to /Applications/Xcode-14.0.1.app
(4/6) Moving Xcode archive Xcode-14.0.1+14A400.xip to the Trash
(5/6) Checking security assessment and code signing
(6/6) Finishing installation
xcodes requires superuser privileges in order to finish installation.
macOS User Password: 

Xcode 14.0.1 has been installed to /Applications/Xcode-14.0.1.app
```

Reply:
ah, i see now.. can we refactor that a bit. 

Can we remove the `exitAfterInstall` property on the `installer.install`. It's a bit confusing on doing it there if in the end we are using that to skip the select

Instead can we pass the select option inside this function and the last step check that, selecting if user wants.

Reply:
I refactored this.